src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private Object[] arrayOfObjects;
+    private int size;
+
+    public ArrayList() {
+        arrayOfObjects = new Object[DEFAULT_SIZE];
+    }
 
     @Override
     public void add(T value) {
+        if (size == arrayOfObjects.length) {
+            growCapacity();
+        }
 
+        arrayOfObjects[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bound");
+        }
+
+        int arrayLength = arrayOfObjects.length;
+        Object[] tempo = new Object[arrayLength + 1];
+        System.arraycopy(arrayOfObjects, 0, tempo, 0, arrayLength);
+        arrayOfObjects = new Object[arrayLength + 1];
+        arrayOfObjects[index] = value;
+        System.arraycopy(tempo, 0, arrayOfObjects, 0, index);
+        System.arraycopy(tempo, index, arrayOfObjects,
+                index + 1, size - index);
 
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < size && index >= 0) {
+            return (T) arrayOfObjects[index];
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("This index doesn't exist");
+        }
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index < size && index >= 0) {
+            arrayOfObjects[index] = value;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("This index doesn't exist");
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Object objectByIndex;
+        if (index >= 0 && index < size) {
+            objectByIndex = arrayOfObjects[index];
+            Object[] temp = new Object[size];
+            System.arraycopy(arrayOfObjects, 0, temp, 0, size);
+            arrayOfObjects = new Object[size - 1];
+            System.arraycopy(temp, 0, arrayOfObjects,
+                    0, index);
+            System.arraycopy(temp, index + 1, arrayOfObjects,
+                    index, size - index - 1);
+            size--;
+            return (T) objectByIndex;
```

Comment:
let's remove the `temp` variable here and use `System.arraycopy` only once
we can move part of the elements inside the same `arrayOfObjects` array, and we don't need to decrease its size